### ISSUE TITLE
Support for mocking Process.clock_gettime via Timecop

### ISSUE THREAD

**Issue by chrisbutcher** [travisjeffery/timecop#220]
It would be great to be able to freeze and travel time returned by this method.

https://ruby-doc.org/core-2.4.2/Process.html#method-c-clock_gettime

More on monotonic clocks: https://www.softwariness.com/articles/monotonic-clocks-windows-and-posix/

**Comment by evan-stripe** [comment]
(It'd be helpful if this worked on CLOCK_REALTIME as well as CLOCK_MONOTONIC - we're moving some particularly performance-sensitive code to use `Process.clock_gettime(Process::CLOCK_REALTIME)` instead of `Time.now` to avoid an object allocation)

**Comment by joshuacronemeyer** [comment]
i've got to spend some time with @wishdev's PR, play around with it and do some tests but think this will be a nice new feature to have in our next release.

**Comment by joshuacronemeyer** [comment]
thanks @adfoster-r7 this slipped off my radar. The PR never had passing tests. I haven't played with it either to see how it works.

**Comment by ecerulm** [comment]
This was closed as completed but still it's not implemented as of timecop 0.9.8 . I checked with 

```ruby
  def test_process_clock_gettime_fail
    time1 = Time.now
    Timecop.freeze(time1) do
      time2 = Process.clock_gettime(Process::CLOCK_MONOTONIC)
      sleep 1
      time3 = Process.clock_gettime(Process::CLOCK_MONOTONIC)
      assert_equal time2, time3
    end
  end
```

The time obtained via `Process.clock_gettime` is not influenced by timecop. 

@joshuacronemeyer , is this issue closed because it's decided that timecop will never support `Process.clock_gettime`? 

Can I suggest to mention README.md in the homepage directly that `Process.clock_gettime` is out of the scope? 

> It provides a unified method to mock Time.now, Date.today, and DateTime.now in a single call (NOTE: Process.clock_gettime is not mocked by timecop)

**Comment by joshuacronemeyer** [comment]
@ecerulm thanks for checking in on this. I just closed this because it was old and nobody was working on it. Not because we're not open to this functionality.

I don't know much about it. I see we had this PR that was here before I started working on the project: https://github.com/travisjeffery/timecop/pull/258

I think we just need somebody to champion a properly tested, working PR.
